src/error.rs

```diff
@@ -57,6 +58,10 @@ pub enum EvalError<'tcx> {
     ExpectedDropGlue(Function<'tcx>),
     ManuallyCalledDropGlue,
     Panic,
+    NonZstToZstCast(PrimVal),
+    ZstToNonZstCast(Ty<'tcx>),
+    ZstToPtrComparison,
+    ReadFromZst,
```

Comment:
All these new error cases make `PrimVal::Zst` seem less clean than the previous state.